src/hooks/useCodapState.tsx

```diff
@@ -43,6 +43,12 @@ export interface InteractiveState {
   activeTableIndex: number;
 }
 
+function unprotectedCollectionsModel(snap?: CollectionsModelSnapshot) {
+  const collectionsModel = CollectionsModel.create(snap);
+  unprotect(collectionsModel);
```

Comment:
This does seems a bit sketchy, but I don't have a better idea other than perhaps not using MST for this purpose. At the very least a comment here would be worthwhile.

Reply:
I'm not really sure why `unprotect` was used here instead of just adding actions to the model that accomplish the necessary mutations.